app/assets/javascripts/rooms/show/components/change_task/change_task.coffee

```diff
@@ -19,20 +23,37 @@ class @ChangeTask
   remove_confimation_vote = ->
     $('.card.selectable').removeClass('confirmed')
 
+  free_all_cards_from_users = ->
+    $('.finish-label').hide()
+    $('.page-header').data('room-status', 'in_progress')
+    $('.card.selectable').not('.result').find('.inner').css('background', '#161ec9')
+    $('.card.selectable').not('.result').css('border-color', '#161ec9')
+
   handler_events: ->
     handle_on_change_task_click.call(@)
 
   handle_on_change_task_click = ->
     ctc = @change_task_channel
-    $('.change-task .item').on 'click', ->
-      change_active_class_from_change_task.call(@)
-      task_id = $(@).data('id')
-      ctc.changeTask({ title: @.innerHTML, task_id: task_id })
-
-  change_active_class_from_change_task = ->
-    $('.change-task .item').removeClass('active')
-    $(@).addClass('active')
+    $('#start_votation').on 'click', ->
+      remove_select_from_select_task()
+      reset_votation()
+      selected = $('.task-select').find('option:selected')
+      $('.task-select')
+      if task_id = selected.val()
+        ctc.changeTask({ title: selected.html(), task_id: task_id })
+      $('#start_votation_modal').modal('toggle')
+
+  reset_votation = ->
+    $('span.steps').each ->
+      @.voted = undefined
+      $('.steps').html('?')
+
+  remove_select_from_select_task = ->
+    $('.task-select').val('')
+    $('.filter-option').html('Nothing Selected')
+    $('.dropdown-menu .inner li').removeClass('selected')
 
 $(document).on 'turbolinks:load', ->
```

Comment:
Is it a pattern to use an initialization of the DOM inside the Component classes? `$(document).on ...`

I think can be hard to maintain and to know where is initialize the pages and components correctly.

Reply:
No is not a pattern I will put in the main file from page folder